starlette/testclient.py

```diff
@@ -352,7 +352,7 @@ async def _asgi_send(self, message: Message) -> None:
 
     def _raise_on_close(self, message: Message) -> None:
         if message["type"] == "websocket.close":
-            raise WebSocketDisconnect(message.get("code", 1000))
+            raise WebSocketDisconnect(message.get("code", 1000), message.get("reason"))
```

Comment:
I think this should default to empty string, but at the end it doesn't make a difference. Your call.
```suggestion
            raise WebSocketDisconnect(message.get("code", 1000), message.get("reason", ""))
```

Reply:
I noticed that earlier and changed `WebsocketDisconnect`:

```python
class WebSocketDisconnect(Exception):
     def __init__(self, code: int = 1000, reason: str = None) -> None:
         self.code = code
         self.reason = reason or ""
```

I think this will do that, right?

Reply:
Yeah, but the raise on the line that I've commented sends either the reason or None. In theory, `WebSocketDisconnect` doesn't receive `None` (looking at the annotation).

Reply:
The annotation is right.

Reply:
Sorry I want to ask this out of curiosity, not for this PR.I get your point that in this internal code we will fix this.
But I guess both Python and MyPy (strict) will be ok about passing `None` to `reason`, right?

Reply:
If you mean the `strict` flag, then no. See below:

```bash
❯ mypy --strict main.py      
main.py:1: error: Incompatible default for argument "b" (default has type "None", argument has type "str")
main.py:4: error: Argument 1 to "a" has incompatible type "None"; expected "str"
Found 2 errors in 1 file (checked 1 source file)
(base) 
~/Development via 🐍 v3.8.8 on ☁️  marcelotryle@gmail.com 
❯ cat main.py                                                          
def a(b: str = None) -> None:
    return None

a(None)
```

Reply:
(we don't use the strict flag on starlette, jfyk)

Reply:
But this will be ok:

```python
import typing

def a(b: typing.Optional[str] = None) -> None:
    return None

a(None)
```

I'll fix this as it's not probably related and we can discuss later. Thank you.